api/src/main/java/org/infinispan/api/search/reactive/QueryParameters.java

```diff
@@ -0,0 +1,27 @@
+package org.infinispan.api.search.reactive;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class QueryParameters {
```

Comment:
`QueryParameters` should be an interface. Then you'd have a public+final implementation that's non empty. This way, you can easily define an empty query parameters implementation of the interface.

Reply:
ok

Reply:
How would the API look like if it's an interface instead an object?

Reply:
What is your doubt exactly? 

Here's some quickly written code:

```
interface QueryParameters {

    QueryParameters append(String name, Object value);
    Map<String, Object> asMap();

    public static final QueryParameters of(String name, Object value) {
       ... 
       // create a private instance of a non-empty query parameters implementation, e.g. `QueryParametersImpl`
    }

    final class Empty implements QueryParameters {
        QueryParameters append(String name, Object value) {
            return this;
        }

        Map<String, Object> asMap() {
            return Collections.emptyMap();
        }
    }
}
```

Also, I'd favour a static initializer called `of` rather than `init`.

You could make QueryParameters.Empty exterrnal to the interface and make it private, and then you could have two static methods in QueryParameters interface, `of` and `empty`:

```
interface QueryParameters {
    public static final QueryParameters of(String name, Object value);
    public static final QueryParameters empty();
}
```

Reply:
@galderz I wouldn't want the API to allow `QueryParameters.empty().append(a, b)`, if we want `of` and `empty` methods we should put `append` in a separate builder interface. 

TBH I'm not sure we need a class/interface at all, we could have just `withQueryParameters(String, Object)` and `withQueryParameters(Map<String, Object>)`.

Reply:
Ok! I prefect the of as you suggest! Let me modify

Reply:
In the new api I added ```withQueryParameter``` that makes easy to pass query parameters in a fluent way. I did this class before the method, and I imagined a way to create the parameters in a fluent way instead a hashmap. but thinking about it twice and with Java 11 in mind, I think Dan's suggestion is more than enough